### PR TITLE
feat: Theme Patch

### PR DESCRIPTION
**Pull request by Sculas**
Adds a Theme Patch and deprecates the Amoled Patch.

Closes https://github.com/orgs/revanced/discussions/121

### CODE REVIEW

**Comment by Sculas** [comment]
If someone has the time to test this PR that would be very appreciated since I'm currently unable to.

File: src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt:33]:
```suggestion
                val element = node as Element
```

File: src/main/kotlin/app/revanced/patches/youtube/layout/theme/Themes.kt
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/theme/Themes.kt:3]: To make it clearer where the class Themes belongs, make it a child class of the `theme` patch.  The usage would be `ThemePatch.Themes.Amoled`.

**Review by oSumAtrIX — COMMENTED** [review]
Looks good

File: src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt:20]: While using the annotation directly here works, it would be better to be consistent with the other patches. See `AmoledCompatibility`.

File: src/main/kotlin/app/revanced/patches/youtube/layout/theme/Themes.kt
Comment (Sculas) [reply src/main/kotlin/app/revanced/patches/youtube/layout/theme/Themes.kt:3]: Sure, will do

File: src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt
Comment (Sculas) [reply src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt:20]: I did it because it's not shared, but for consistency I can understand why. I'll do that then

Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt:20]: Yeah theres multiple patches where a separate annotation is technically not needed, it's soley for consistency.

**Comment by OxrxL** [comment]
@Sculas 

Can you eventually add my patch to change filtersbar comments color to this PR?

**Comment by oSumAtrIX** [comment]
@OxrxL please link to what you mean by filtersbar.

**Comment by OxrxL** [comment]
> @OxrxL please link to what you mean by filtersbar.

https://github.com/revanced/revanced-patches/issues/234

**Comment by Sculas** [comment]
> Can you eventually add my patch to change filtersbar comments color to this PR?

I think that's out of scope for this PR, but for a different one, sure.

**Comment by OxrxL** [comment]
> > Can you eventually add my patch to change filtersbar comments color to this PR?
> 
> I think that's out of scope for this PR, but for a different one, sure.

Being an amoled patch, i thought it could be merged together (also if it's a bytecode patch).

**Comment by oSumAtrIX** [comment]
The fix requires changes to bytecode, for that the enum would need a different structure to also allow changes to the bytecode, which shouldn't oppose much problems in a future PR.

File: src/main/kotlin/app/revanced/patches/youtube/layout/theme/Themes.kt
Comment (Sculas) [reply src/main/kotlin/app/revanced/patches/youtube/layout/theme/Themes.kt]: Done: 937201a85cc92c8683ee730fad5f45279200b503

**Comment by Sculas** [comment]
This PR should be good to merge now, waiting for approval.

**Review by oSumAtrIX — APPROVED** [review]

**Pull request merged**
